packages/lisk-p2p/src/p2p.ts

```diff
@@ -321,27 +336,10 @@ export class P2P extends EventEmitter {
 		});
 	}
 
-	private async _stopHTTPServer(): Promise<void> {
-		return new Promise<void>(resolve => {
-			this._httpServer.close(() => {
-				this._isActive = false;
-				resolve();
-			});
-		});
-	}
-
-	private async _stopWSServer(): Promise<void> {
-		return new Promise<void>(resolve => {
-			this._scServer.close(() => {
-				this._isActive = false;
-				resolve();
-			});
-		});
-	}
-
-	private async _stopPeerServer(): Promise<void> {
-		await this._stopWSServer();
-		await this._stopHTTPServer();
+	private _stopPeerServer(): void {
+		this._scServer.close();
```

Comment:
we don't need to wait for the callback to make sure?

Reply:
Ah yes we did. I confused this issue for something else. There was a legitimate reason why it wasn't shutting down properly.